support/src/benchmark.rs

```diff
@@ -0,0 +1,35 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::traits::GetWorstCase;
+
+/// Manly used for the DIP benchmarking. The worst case can be created for the did and submitter.
+pub struct IdentityContext<DidIdentifier, AccountId> {
```

Comment:
Yes, but this is not easy. If I move it to runtime-common, the pallet-dip-consumer and pallet-dip-provider will have a dependency on runtime common. If I remember correctly, we intended to avoid dependencies between runtime common and the pallets.

if I move it to the kilt-dip-support crate, I encounter a circular dependency between the crate and the pallets.

Reply:
Mh I see. Maybe `kilt-dip-support` should not depend on the pallets. Ok I guess for now we can leave it in here then 👍 As long as the module is feature-gated, it's not a big issue.